events-api/src/modules/events/events.repository.ts

```diff
@@ -1,21 +1,120 @@
 import { db } from "#/database/database";
 import { EventModel } from "#/modules/events/events.model";
+import { eventColumns } from "#/shared/utils";
+import { TicketModel } from "../tickets/tickets.model";
+import { EventCreateDto } from "./dto/requests/events-create.dto";
+import { v4 } from "uuid";
 
 export class EventsRepository {
   static async getAll(): Promise<EventModel[]> {
     const result = db.execute<EventModel>(`
         SELECT
-            event_id as eventId,
-            event_name as eventName,
-            event_description as eventDescription,
-            event_location as eventLocation,
-            event_date as eventDate,
-            event_created_at as eventCreatedAt,
-            event_updated_at as eventUpdatedAt
+            ${eventColumns}
         FROM
             events
     `);
 
     return result;
   }
+
+  static async findById(id: string): Promise<EventModel[] | null> {
+    const res = await db.execute<EventModel>(
+      `SELECT ${eventColumns} FROM events WHERE event_id = ?`,
+      [id]
+    );
+
+    if (res.length === 0) {
+      return null;
+    }
+
+    return res;
```

Comment:
Return singular record, not an array of the records

Reply:
![image](https://github.com/user-attachments/assets/22e436cc-9f87-4a56-9dcd-ecdac1c38de0)

first checking by id if there is event_id then I will perform all other operations

Reply:
Yes, but you are returning an array, not a single object